lib/game/components/asteroid_component.dart

```diff
@@ -0,0 +1,155 @@
+import 'dart:math';
+
+import 'package:flame/collisions.dart';
+import 'package:flame/components.dart';
+import 'package:flutter/material.dart';
+import 'package:unicorn_crossing/game/components/laser_component.dart';
+
+class AsteroidComponent extends PositionComponent with CollisionCallbacks {
+  AsteroidComponent({
+    required Vector2 position,
+    required Vector2 size,
+    required double speed,
+  })  : _speed = speed,
+        super(
+          position: position,
+          size: size,
+        );
+
+  final double _speed;
+  final Random _random = Random();
+  final _hitbox = RectangleHitbox();
+  late List<_Pixel> _pixels;
+  bool _isExploding = false;
+
+  @override
+  Future<void> onLoad() async {
+    await super.onLoad();
+    add(_hitbox); // Add a collision shape
+
+    // Generate the pixelated blob shape
+    _generatePixelatedShape();
+  }
+
+  void _generatePixelatedShape() {
```

Comment:
We should probably use the particle system that flame provide, and this could be a component of its own, something like a `ExplosionComponent`.